### PR TITLE
Bucket index updater should ignore meta not found for partial blocks

### PR DESCRIPTION
**Pull request by alexqyle**
**What this PR does**:
Bucket index updater should ignore meta not found for partial blocks to avoid increasing `thanos_objstore_bucket_operation_failures_total` metric when checking partial blocks.

**Checklist**
- [ ] Tests updated
- [ ] Documentation added
- [x] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

**Review by alanprot — APPROVED** [review]

**Comment by alanprot** [comment]
LGTM! Thanks

File: pkg/storage/tsdb/bucketindex/updater.go [hunk 1]
```diff
@@ -123,7 +123,7 @@ func (w *Updater) updateBlockIndexEntry(ctx context.Context, id ulid.ULID) (*Blo
 	metaFile := path.Join(id.String(), block.MetaFilename)
 
 	// Get the block's meta.json file.
-	r, err := w.bkt.Get(ctx, metaFile)
+	r, err := w.bkt.ReaderWithExpectedErrs(w.bkt.IsObjNotFoundErr).Get(ctx, metaFile)
```
Comment (alvinlin123) [inline pkg/storage/tsdb/bucketindex/updater.go:126]: Should we have a test on this?

Comment (alexqyle) [reply pkg/storage/tsdb/bucketindex/updater.go:126]: Let me see if it is easy to setup `metricBucket` in the test to make sure the metric not increased in this case.

Comment (alexqyle) [reply pkg/storage/tsdb/bucketindex/updater.go:126]: Added test

**Review by alvinlin123 — APPROVED** [review]

**Pull request merged**